artifactory/commands/golang/go.go

```diff
@@ -139,6 +167,41 @@ func (gc *GoCommand) Run() error {
 
 	return err
 }
+func copyGoPackageFiles(destPath, packageName, rtTargetRepo string, rtDetails auth.ServiceDetails) error {
```

Comment:
This function does a lot more than what its name implies (copyGoPackageFiles) - yes, it indeed copies the go package files, but it also builds the request to get the version from Artofactory. I suggest splitting this function into smaller building blocks.
Having smaller building blocks also makes the code testable. For example, the flow of creating the the request to Artifactory can (and probably should) be unit-tested, possible using table testing.